.github/workflows/backend-ecr.yaml

```
@@ -0,0 +1,56 @@
+name: Build and push backend docker image 
+
+on:
+  push:
+    branches: [ "main"]
+    paths:
+      - 'backend/**'
+      - 'terraform/**'
+      - '.github/workflows/backend-ecr.yaml'
+
+env:
+  AWS_REGION: ${{ vars.AWS_REGION || 'us-east-1' }}
+  ECR_REPOSITORY: ${{ vars.ECR_REPOSITORY }}
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_OIDC_ROLE }}
+          
+      - name: Login to ECR
+        id: ecr
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Build and push image (SHA tag)
+        env:
+          ECR_REGISTRY: ${{ steps.ecr.outputs.registry }}
+        run: |
+          SHORT_SHA="sha-${GITHUB_SHA::7}"
+          IMAGE="$ECR_REGISTRY/${{ env.ECR_REPOSITORY }}"
+          TAGS="$IMAGE:$SHORT_SHA"
+          
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            TAGS="$TAGS,$IMAGE:latest"
+          fi
+
+          echo "Building and pushing $IMAGE with tags: $TAGS"
+          docker build -t "$IMAGE:$SHORT_SHA" -f backend/Dockerfile backend
+          docker push "$IMAGE:$SHORT_SHA"
+          
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            docker tag "$IMAGE:$SHORT_SHA" "$IMAGE:latest"
+            docker push "$IMAGE:latest"
+          fi
+
+          echo "Image pushed: $IMAGE:$SHORT_SHA"
```

```suggestion

          echo "Building and pushing $IMAGE:$SHORT_SHA"
          docker build -t "$IMAGE:$SHORT_SHA" -f backend/Dockerfile backend
          docker push "$IMAGE:$SHORT_SHA"
          
          echo "Pushed: $IMAGE:$SHORT_SHA"
          
          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
            echo "Tagging $IMAGE:$SHORT_SHA as :latest"
            docker tag "$IMAGE:$SHORT_SHA" "$IMAGE:latest"
            docker push "$IMAGE:latest"
            echo "Pushed: $IMAGE:latest"
          fi
```

I believe I'm following the intended logic (always push `:SHA` to include feature branches, update `:latest` on merge), it just took a few rereads with the way the variables are used. Open to alternatives.

---

That is the intended logic. Your suggestion is easier to read!
One thing that might be a concern are rollbacks. They could re-run the last merge into main, is this something worth addressing?